packages/input-datepicker/src/LionInputDatepicker.js

```diff
@@ -280,6 +290,18 @@ export class LionInputDatepicker extends ScopedElementsMixin(OverlayMixin(LionIn
     `;
   }
 
+  _setupOverlayCtrl() {
+    super._setupOverlayCtrl();
+
+    this._overlayCtrl.addEventListener('before-show', () => {
+      if (window.innerWidth >= 600) {
+        this._overlayCtrl.updateConfig(this._defineOverlayConfig());
```

Comment:
I feel like the configuration of the overlay is too spread out across different parts. I would add the `window.innerWidth` switch to `_defineOverlayConfig()`:

```js
_defineOverlayConfig() {
  if (window.innerWidth >= 600) {
    return {
      ...withModalDialogConfig(),
      hidesOnOutsideClick: true,
      ...super._defineOverlayConfig(),
      popperConfig: {
        ...super._defineOverlayConfig().popperConfig,
        placement: 'bottom',
      },
    };
  } else {
    return withBottomSheetConfig();
  }
}
```

and

```js
this._overlayCtrl.addEventListener('before-show', () => {
  this._overlayCtrl.updateConfig(this._defineOverlayConfig());
}
```

Reply:
changed